src/domainobjectfactories/trade_factory.py

```diff
@@ -0,0 +1,228 @@
+            10 character random string representing trade contract id
+        """
+        return self.create_random_string(10)
+
+    @staticmethod
+    def __create_trade_lifecycle_dates():
+        """ return datetime objects representing when the trade was booked,
+        executed and due to be settled respectively. These are currently hard
+        coded to be the current date for booking and executing, and 2 days in
+        the future for settlement. Datetimes are in UTC, and value date is set
+        to 1 minute past midnight on the morning of T+2
+
+        Returns
+        -------
+        Datetime
+            Datetime object in UTC representing booking datetime - set to use
+            current datetime
+        Datetime
+            Datetime object in UTC representing trade datetime - set to use
+            current datetime
+        Datetime
+            Datetime object in UTC representing expected value datetime - set
+            to 1 minute past midgnight on the morning of T+2, where the
+            current date is T
+        """
+        booking_datetime = trade_datetime = datetime.now(timezone.utc)
+        day_after_tomorrow = datetime.now(timezone.utc) + timedelta(days=2)
+        value_datetime = day_after_tomorrow.replace(
+            hour=0, minute=1, second=0, microsecond=0
+        )
+        return booking_datetime, trade_datetime, value_datetime
+
+    def __create_order_id(self):
+        """ Return id of the order
+        Returns
+        -------
+        Integer
+            random integer between 1 and 10000
+        """
+        return self.create_random_integer()
+
+    def __get_account_id(self):
+        """ Return the account id value of an account persisted in the
+        database that is type 'Client' or 'Firm'
+
+        Returns
+        -------
+        String
+            account id of 'Client' or 'Firm' type account from database
+        """
+        account = self.get_random_record_with_valid_attribute(
+            'accounts', 'account_type', ['Counterparty', 'Depot']
+        )
+        return account['account_id']
+
+    def __get_counterparty_id(self):
+        """ Return the account id value of an account persisted in the
+        database that is type 'Counterparty'
+
+        Returns
+        -------
+        String
+            account id of 'Counterparty' type account from database
+        """
+        account = self.get_random_record_with_valid_attribute(
+            'accounts', 'account_type', ['Client', 'Firm', 'Depot']
+        )
+        return account['account_id']
+
+    def __create_trader_id(self):
+        """ Return id of the trader
+        Returns
+        -------
+        String
+            10 character random string representing trader id
+        """
+        return self.create_random_string(10)
+
+    def __create_price(self, quantity):
+        """ Return total value of the trade, found by multiplying instrument
+        quantity by a rnadomly generated unit price
```

Comment:
typo.  Also very briefly say that we're generating the unit price because the trade isn't guaranteed to have been done at the standard market price